src/mutations/appSetting.ts

```diff
@@ -14,27 +14,38 @@ import { AppSetting, convertJs } from '@graasp/sdk';
 import { AppSettingRecord } from '@graasp/sdk/frontend';
 
 export default (queryClient: QueryClient, queryConfig: QueryClientConfig) => {
-  const { notifier } = queryConfig;
+  const { notifier, enableWebsocket } = queryConfig;
 
   queryClient.setMutationDefaults(MUTATION_KEYS.POST_APP_SETTING, {
     mutationFn: (payload: Partial<AppSetting>) => {
       const apiHost = getApiHost(queryClient);
       const data = getDataOrThrow(queryClient);
       return Api.postAppSetting({ ...data, body: payload, apiHost });
     },
-    onSuccess: (newData: AppSetting) => {
+    onSuccess: (newAppSetting: AppSetting) => {
       const { itemId } = getData(queryClient);
       const key = buildAppSettingsKey(itemId);
       const prevData = queryClient.getQueryData<List<AppSettingRecord>>(key);
-      queryClient.setQueryData(key, prevData?.push(convertJs(newData)));
+      const newData: AppSettingRecord = convertJs(newAppSetting);
+      if (enableWebsocket) {
+        // check that the websocket event has not already been received and therefore the data were added
+        if (prevData?.findIndex((a) => a.id === newData.id) === -1) {
+          queryClient.setQueryData(key, prevData?.push(newData));
+        }
+      } else {
+        // No websockets, then updates the data.
+        queryClient.setQueryData(key, prevData?.push(newData));
+      }
```

Comment:
Same question on data structure. You can simplify as above by removing the websocket check, and consider taking care of `prevData` being nullable as before.